gibberish_server/models/room.js

```diff
@@ -28,21 +28,25 @@ const ROUND_NUMBER = 10;
 Object.freeze(ROUND_NUMBER);
 
 class Room {
-    constructor(id = uid(), state = STATE.GAME_WAITING, round = 0, players = [], startedTime = null, qna = this.generateQuestions()) {
+    constructor(id = uid(), state = STATE.GAME_WAITING, round = 0, players = [], startedTime = null, qna = this.generateQuestions(), oldQna = new Set()) {
         this.id = id;
         this.state = state;
         this.round = round;
         this.players = players;
         this.startedTime = startedTime;
         this.qna = qna
+        this.oldQna = oldQna;
         this.timer = Math.floor((new Date() - new Date(this.startedTime))/1000)
     }
 
     generateQuestions() {
         let qna = new Set()
         while(qna.size < ROUND_NUMBER) {
             const r = Math.floor(Math.random() * questions.length)
-            qna.add(questions[r])
+            const randomQuestion = questions[r];
+            if (!this.oldQna || (this.oldQna && !this.oldQna.has(randomQuestion))) {
```

Comment:
what does !this.oldQna equate to? does an empty set mean that its false?

Reply:
`!this.oldQna` checks for whether its undefined! Because this function is called inside the constructor as well (line 31), even before `this.oldQna` is set to a new empty Set. So this check is needed for the first game!

Reply:
hmmm i think u can do the restart without having a this.oldQna. Couldn't you just compare with the original this.qna to make sure its not repeated?

Reply:
Oh i wanted to do this initially too! But then there would still be repeats if the room restarts more than twice :/

Reply:
ahhh i see what ure doing, ok coolz